### PR TITLE
feat: Dioxus 0.5 support

### PR DESCRIPTION
**Pull request by marc2332**
Prepare support for Dioxus 0.5

Closes https://github.com/dioxus-community/dioxus-free-icons/issues/26
Closes https://github.com/dioxus-community/dioxus-free-icons/issues/28
Closes https://github.com/dioxus-community/dioxus-free-icons/issues/14

### CODE REVIEW

**Comment by marc2332** [comment]
Looks like material design icons are failed to be written, looking into it

**Comment by iynaix** [comment]
Trying to use it with the following in `Cargo.toml`:

```toml
dioxus-free-icons = { git = "https://github.com/dioxus-community/dioxus-free-icons", branch = "feat/dioxus-0.5", features = [
    "material-design-icons-editor",
] }
```

I'm getting hundreds of errors such as:
```console
error[E0119]: conflicting implementations of trait `IconShape` for type `md_editor_icons::MdMaterialDesignIcons`
    --> /persist/cache/home/iynaix/.local/share/.cargo/git/checkouts/dioxus-free-icons-d246d2ebe6d920da/58f601c/packages/lib/src/icons/md_editor_icons.rs:1816:1
     |
6    | impl IconShape for MdMaterialDesignIcons {
     | ---------------------------------------- first implementation here
...
1816 | impl IconShape for MdMaterialDesignIcons {
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `md_editor_icons::MdMaterialDesignIcons`

error[E0119]: conflicting implementations of trait `IconShape` for type `md_editor_icons::MdMaterialDesignIcons`
    --> /persist/cache/home/iynaix/.local/share/.cargo/git/checkouts/dioxus-free-icons-d246d2ebe6d920da/58f601c/packages/lib/src/icons/md_editor_icons.rs:1837:1
     |
6    | impl IconShape for MdMaterialDesignIcons {
     | ---------------------------------------- first implementation here
...
1837 | impl IconShape for MdMaterialDesignIcons {
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `md_editor_icons::MdMaterialDesignIcons`

error[E0119]: conflicting implementations of trait `IconShape` for type `md_editor_icons::MdMaterialDesignIcons`
    --> /persist/cache/home/iynaix/.local/share/.cargo/git/checkouts/dioxus-free-icons-d246d2ebe6d920da/58f601c/packages/lib/src/icons/md_editor_icons.rs:1858:1
     |
6    | impl IconShape for MdMaterialDesignIcons {
     | ---------------------------------------- first implementation here
...
1858 | impl IconShape for MdMaterialDesignIcons {
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `md_editor_icons::MdMaterialDesignIcons`

Some errors have detailed explanations: E0119, E0428.
For more information about an error, try `rustc --explain E0119`.
error: could not compile `dioxus-free-icons` (lib) due to 595 previous errors
warning: build failed, waiting for other jobs to finish...
```

Am I missing something obvious here like having to run a codegen step?

**Comment by marc2332** [comment]
> Trying to use it with the following in `Cargo.toml`:
> 
> ```toml
> dioxus-free-icons = { git = "https://github.com/dioxus-community/dioxus-free-icons", branch = "feat/dioxus-0.5", features = [
>     "material-design-icons-editor",
> ] }
> ```
> 
> I'm getting hundreds of errors such as:
> 
> ```
> error[E0119]: conflicting implementations of trait `IconShape` for type `md_editor_icons::MdMaterialDesignIcons`
>     --> /persist/cache/home/iynaix/.local/share/.cargo/git/checkouts/dioxus-free-icons-d246d2ebe6d920da/58f601c/packages/lib/src/icons/md_editor_icons.rs:1816:1
>      |
> 6    | impl IconShape for MdMaterialDesignIcons {
>      | ---------------------------------------- first implementation here
> ...
> 1816 | impl IconShape for MdMaterialDesignIcons {
>      | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `md_editor_icons::MdMaterialDesignIcons`
> 
> error[E0119]: conflicting implementations of trait `IconShape` for type `md_editor_icons::MdMaterialDesignIcons`
>     --> /persist/cache/home/iynaix/.local/share/.cargo/git/checkouts/dioxus-free-icons-d246d2ebe6d920da/58f601c/packages/lib/src/icons/md_editor_icons.rs:1837:1
>      |
> 6    | impl IconShape for MdMaterialDesignIcons {
>      | ---------------------------------------- first implementation here
> ...
> 1837 | impl IconShape for MdMaterialDesignIcons {
>      | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `md_editor_icons::MdMaterialDesignIcons`
> 
> error[E0119]: conflicting implementations of trait `IconShape` for type `md_editor_icons::MdMaterialDesignIcons`
>     --> /persist/cache/home/iynaix/.local/share/.cargo/git/checkouts/dioxus-free-icons-d246d2ebe6d920da/58f601c/packages/lib/src/icons/md_editor_icons.rs:1858:1
>      |
> 6    | impl IconShape for MdMaterialDesignIcons {
>      | ---------------------------------------- first implementation here
> ...
> 1858 | impl IconShape for MdMaterialDesignIcons {
>      | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `md_editor_icons::MdMaterialDesignIcons`
> 
> Some errors have detailed explanations: E0119, E0428.
> For more information about an error, try `rustc --explain E0119`.
> error: could not compile `dioxus-free-icons` (lib) due to 595 previous errors
> warning: build failed, waiting for other jobs to finish...
> ```
> 
> Am I missing something obvious here like having to run a codegen step?


No, it looks like the codegen generated is wrong, will check asap! Thanks for the report!

**Comment by marc2332** [comment]
> Trying to use it with the following in `Cargo.toml`:
> 
> ```toml
> dioxus-free-icons = { git = "https://github.com/dioxus-community/dioxus-free-icons", branch = "feat/dioxus-0.5", features = [
>     "material-design-icons-editor",
> ] }
> ```
> 
> I'm getting hundreds of errors such as:
> 
> ```
> error[E0119]: conflicting implementations of trait `IconShape` for type `md_editor_icons::MdMaterialDesignIcons`
>     --> /persist/cache/home/iynaix/.local/share/.cargo/git/checkouts/dioxus-free-icons-d246d2ebe6d920da/58f601c/packages/lib/src/icons/md_editor_icons.rs:1816:1
>      |
> 6    | impl IconShape for MdMaterialDesignIcons {
>      | ---------------------------------------- first implementation here
> ...
> 1816 | impl IconShape for MdMaterialDesignIcons {
>      | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `md_editor_icons::MdMaterialDesignIcons`
> 
> error[E0119]: conflicting implementations of trait `IconShape` for type `md_editor_icons::MdMaterialDesignIcons`
>     --> /persist/cache/home/iynaix/.local/share/.cargo/git/checkouts/dioxus-free-icons-d246d2ebe6d920da/58f601c/packages/lib/src/icons/md_editor_icons.rs:1837:1
>      |
> 6    | impl IconShape for MdMaterialDesignIcons {
>      | ---------------------------------------- first implementation here
> ...
> 1837 | impl IconShape for MdMaterialDesignIcons {
>      | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `md_editor_icons::MdMaterialDesignIcons`
> 
> error[E0119]: conflicting implementations of trait `IconShape` for type `md_editor_icons::MdMaterialDesignIcons`
>     --> /persist/cache/home/iynaix/.local/share/.cargo/git/checkouts/dioxus-free-icons-d246d2ebe6d920da/58f601c/packages/lib/src/icons/md_editor_icons.rs:1858:1
>      |
> 6    | impl IconShape for MdMaterialDesignIcons {
>      | ---------------------------------------- first implementation here
> ...
> 1858 | impl IconShape for MdMaterialDesignIcons {
>      | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `md_editor_icons::MdMaterialDesignIcons`
> 
> Some errors have detailed explanations: E0119, E0428.
> For more information about an error, try `rustc --explain E0119`.
> error: could not compile `dioxus-free-icons` (lib) due to 595 previous errors
> warning: build failed, waiting for other jobs to finish...
> ```
> 
> Am I missing something obvious here like having to run a codegen step?

Try now! I think it should work

**Comment by iynaix** [comment]
It seems to work now (no more build errors), but I am running into #26. There's an empty path over the actual icon so it always shows up as a filled box. :(

**Comment by marc2332** [comment]
> It seems to work now (no more build errors), but I am running into #26. There's an empty path over the actual icon so it always shows up as a filled box. :(

Will try to fix asap!

**Comment by marc2332** [comment]
> It seems to work now (no more build errors), but I am running into #26. There's an empty path over the actual icon so it always shows up as a filled box. :(

Try now! I think I got it working 😄

**Comment by iynaix** [comment]
It's working great now! :heart:

**Comment by marc2332** [comment]
> It's working great now! ❤️

Awesome!

File: packages/example/Cargo.toml [hunk 1]
```diff
@@ -4,11 +4,8 @@ version = "0.1.0"
 authors = ["nissy-dev <nd.12021218@gmail.com>"]
 edition = "2021"
 
-# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
-
 [dependencies]
-dioxus = { version = "0.4.0" }
-dioxus-web = { version = "0.4.0" }
+dioxus = { version = "0.5.0-alpha.0", features = ["web"] }
```
Comment (houseme) [inline packages/example/Cargo.toml:8]: 0.5.0

**Pull request merged**